include/zephyr/mgmt/mcumgr/callbacks.h

```diff
@@ -0,0 +1,221 @@
+	MGMT_EVT_OP_CMD_DONE			= MGMT_DEF_EVT_OP_ID(MGMT_EVT_GRP_SMP, 2),
+
+	/** Used to enable all smp_group events. */
+	MGMT_EVT_OP_CMD_ALL			= MGMT_DEF_EVT_OP_ALL(MGMT_EVT_GRP_SMP),
+};
+
+/**
+ * MGMT event opcodes for filesystem management group.
+ */
+enum fs_mgmt_group_events {
+	/** Callback when a file has been accessed, data is fs_mgmt_file_access. */
+	MGMT_EVT_OP_FS_MGMT_FILE_ACCESS		= MGMT_DEF_EVT_OP_ID(MGMT_EVT_GRP_FS, 0),
+
+	/** Used to enable all fs_mgmt_group events. */
+	MGMT_EVT_OP_FS_MGMT_ALL			= MGMT_DEF_EVT_OP_ALL(MGMT_EVT_GRP_FS),
+};
+
+/**
+ * MGMT event opcodes for image management group.
+ */
+enum img_mgmt_group_events {
+	/** Callback when a client sends a file upload chunk, data is img_mgmt_upload_check. */
+	MGMT_EVT_OP_IMG_MGMT_DFU_CHUNK		= MGMT_DEF_EVT_OP_ID(MGMT_EVT_GRP_IMG, 0),
+
+	/** Callback when a DFU operation is stopped. */
+	MGMT_EVT_OP_IMG_MGMT_DFU_STOPPED	= MGMT_DEF_EVT_OP_ID(MGMT_EVT_GRP_IMG, 1),
+
+	/** Callback when a DFU operation is started. */
+	MGMT_EVT_OP_IMG_MGMT_DFU_STARTED	= MGMT_DEF_EVT_OP_ID(MGMT_EVT_GRP_IMG, 2),
+
+	/** Callback when a DFU operation has finished being transferred. */
+	MGMT_EVT_OP_IMG_MGMT_DFU_PENDING	= MGMT_DEF_EVT_OP_ID(MGMT_EVT_GRP_IMG, 3),
+
+	/** Callback when an image has been confirmed. */
+	MGMT_EVT_OP_IMG_MGMT_DFU_CONFIRMED	= MGMT_DEF_EVT_OP_ID(MGMT_EVT_GRP_IMG, 4),
+
+	/** Used to enable all img_mgmt_group events. */
+	MGMT_EVT_OP_IMG_MGMT_ALL		= MGMT_DEF_EVT_OP_ALL(MGMT_EVT_GRP_IMG),
+};
+
+/**
+ * MGMT event opcodes for operating system management group.
+ */
+enum os_mgmt_group_events {
+	/** Callback when a reset command has been received. */
+	MGMT_EVT_OP_OS_MGMT_RESET		= MGMT_DEF_EVT_OP_ID(MGMT_EVT_GRP_OS, 0),
+
+	/** Used to enable all os_mgmt_group events. */
+	MGMT_EVT_OP_OS_MGMT_ALL			= MGMT_DEF_EVT_OP_ALL(MGMT_EVT_GRP_OS),
+};
+
+/**
+ * MGMT callback struct
+ */
+struct mgmt_callback {
+	/** Entry list node. */
+	sys_snode_t node;
+
+	/** Callback that will be called. */
+	mgmt_cb callback;
+
+	/** MGMT_EVT_[...] Event ID for handler to be called on. */
+	uint32_t event_id;
+};
+
+/**
+ * MGMT_EVT_OP_CMD_RECV, MGMT_EVT_OP_CMD_STATUS, MGMT_EVT_OP_CMD_DONE arguments
+ */
+struct mgmt_evt_op_cmd_arg {
+	/** MGMT_GROUP_ID_[...] */
+	uint16_t group;
+
+	/** Message ID within group */
+	uint8_t id;
+
+	union {
+		/** MGMT_ERR_[...], used in MGMT_EVT_OP_CMD_DONE */
+		int err;
+
+		/** IMG_MGMT_ID_UPLOAD_STATUS_[...], used in MGMT_EVT_OP_CMD_STATUS */
+		int status;
+	};
+};
+
+/**
+ * @brief This function is called to notify registered callbacks about mgmt notifications/events.
+ *
+ * @param event		MGMT_EVT_OP_[...].
+ * @param data		Optional event argument.
+ * @param data_size	Size of optional event argument (0 if none).
+ *
+ * @return		MGMT_ERR_[...] either MGMT_ERR_EOK if all handlers returned it, or the
+ *			error code of the first handler that returned an error.
+ */
+int32_t mgmt_notify_callbacks(uint32_t event, void *data, size_t data_size);
```

Comment:
Would you consider changing order to `mgmt_callback_notify`, `mgmt_callback_register` and `mgmt_callback_unregister`.
This would crate `mgmt_callback` namespace and make the functions look like they work around the same area.